### PR TITLE
[Eligibility message] Adding endpoint on controller for updating eligibility message

### PR DESCRIPTION
**Pull request by azhangggggg**
### Description

Adding endpoint on AdminProgramBlockPredicatesController to update eligibility message there. This method takes the value eligibilityMsg from BlockEligibilityMessageForm on the frontend and update the field. The program should be in draft mode.

It will be further tested when the frontend part of eligibility message gets completed.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [ ] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [ ] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Removed the release notes section if the title is sufficient for the release notes description, or put more details in that section.
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary. For user-facing features, consider updating [the user docs](https://github.com/civiform/docs). For "under-the-hood" changes or things more relevant to developers, consider updating [the dev wiki](https://github.com/civiform/civiform/wiki).
- [x] Ensured PII wasn't added to any new logs, unless it was guarded by `isDevOrStaging`

### Issue(s) this completes

Part of #9303

### CODE REVIEW

File: server/app/controllers/admin/AdminProgramBlockPredicatesController.java [hunk 1]
```diff
@@ -379,4 +383,36 @@ public Result destroyEligibility(long programId, long blockDefinitionId) {
...
+        formFactory.form(BlockEligibilityMessageForm.class).bindFromRequest(request);
+    String newMessage = EligibilityMsgform.get().getEligibilityMessage();
+    String toastMessage;
+    String toastType;
+
+    try {
+      programService.setBlockEligibilityMessage(
+          programId, blockDefinitionId, Optional.of(LocalizedStrings.of(Locale.US, newMessage)));
+
+      toastType = "success";
```
Comment (avaleske) [inline server/app/controllers/admin/AdminProgramBlockPredicatesController.java:402]: Can we use the value from the enum?
https://github.com/civiform/civiform/blob/441995d2150ed4fc6f0b1ce46f2adf388e34edb9/server/app/views/components/ToastMessage.java#L18

File: server/app/controllers/admin/AdminProgramBlockPredicatesController.java [hunk 2]
```diff
@@ -379,4 +383,36 @@ public Result destroyEligibility(long programId, long blockDefinitionId) {
...
+      if (newMessage.isBlank()) {
+        toastMessage = "Eligibility message removed.";
+      } else {
+        toastMessage = "Eligibility message set to " + newMessage;
```
Comment (avaleske) [inline server/app/controllers/admin/AdminProgramBlockPredicatesController.java:406]: When the message is very long and includes markdown, what happens to the toast?

**Review by avaleske — CHANGES_REQUESTED** [review]
Small changes but overall lgtm!

File: server/app/controllers/admin/AdminProgramBlockPredicatesController.java (hunk 2 above)
Comment (azhangggggg) [reply server/app/controllers/admin/AdminProgramBlockPredicatesController.java:408]: Good question, I'll test that scenario in browser tests when implementing frontend.

**Review by avaleske — APPROVED** [review]

**Pull request merged**